solver/src/main/java/org/chocosolver/sat/MiniSat.java

```diff
@@ -922,35 +747,45 @@ private int getBacktrackLevel(TIntList out_learnt) {
 
     boolean simplify() {
         assert (trailMarker() == rootlvl);
-
         if (ok_) propagate();
         if (!ok_ || (confl != C_Undef))
             return ok_ = false;
         // TODO
-        /*if (nAssigns() == simpDB_assigns || (simpDB_props > 0))
-            return true;
+//        if (nAssigns() == simpDB_assigns || (simpDB_props > 0))
+//            return true;
 
         // Remove satisfied clauses:
         removeSatisfied(learnts);
-        if (remove_satisfied)        // Can be turned off.
-            removeSatisfied(clauses);
-        */
-        rebuildOrderHeap();
-
-        /*simpDB_assigns = nAssigns();
-        simpDB_props = clauses_literals + learnts_literals;   // (shouldn't depend on stats really, but it will do for now)
-        */
+//        if (remove_satisfied)        // Can be turned off.
+//            removeSatisfied(clauses);
+
         return true;
     }
 
-    private void rebuildOrderHeap() {
-        TIntList vs = new TIntArrayList();
-        for (int v = 0; v < nVars(); v++)
-            if (decision.get(v) && valueVar(v) == lUndef)
-                vs.add(v);
-        order_heap.build(vs);
+    void removeSatisfied(ArrayList<Clause> cs) {
+        int i, j;
+        for (i = j = 0; i < cs.size(); i++) {
+            Clause c = cs.get(i);
+            if (satisfied(c)){
+                removeClause(cs.get(i));
+            }else{
+                cs.set(j++, cs.get(i));
+
+            }
+        }
+//        cs.shrink(i - j);
+        if (cs.size() > j) {
+            cs.subList(j, cs.size()).clear();
```

Comment:
rebase branch and check replacement by .resetQuick()

Reply:
here, `cs.subList` returns a view of cs as a `List` without the `resetQuick()` method.